### ISSUE TITLE
modern C++ library should use modern CMake

### ISSUE THREAD

**Issue by Xeverous** [libSDL2pp/libSDL2pp#131]
If you aren't aware, a bit later than modern C++ happened, *modern CMake* happened. Just like C++ it also kept backwards compatibilty but there are much better ways to do things now. Specifically:

https://github.com/libSDL2pp/libSDL2pp/blob/a02d5a81c3d4122cb578fcd1e5cd4e836878f63b/CMakeLists.txt#L203-L209

This code currently uses old directory-based flags, which do not work well in modern CMake. As of now, any library should be used like this:

```cmake
target_link_libraries(my_project PUBLIC SDL2pp)
```

...**and nothing more**. In modern CMake this single command (with new keywords: `PRIVATE` (build requirement) /`INTERFACE` (usage requirement) / `PUBLIC` (both)) will forward all transitive requirements of the dependency: include paths, compiler options, linker flags and so on. The requirement for this is that the target is also defined this way:

```cmake
add_library(SDL2pp ...)
target_sources(SDL2pp PRIVATE ...) # sources
target_sources(SDL2pp PUBLIC ...) # headers
target_include_directories(SDL2pp PUBLIC ...)
target_compile_options(SDL2pp PRIVATE ...)
target_*(SDL2pp PUBLIC/PRIVATE ...) # and so on...
```

In other words, modern CMake projects do not want variables - they want *targets* they can link to. This is especially true for libraries that may want to use subrepository approach, which can not work in the old system.

I could make a PR for this but note: this will raise minimum required CMake version.
